communicator/config.go

```diff
@@ -503,12 +503,12 @@ func (c *Config) prepareSSH(ctx *interpolate.Context) []error {
 
 		if c.SSHBastionPrivateKeyFile == "" && c.SSHPrivateKeyFile != "" {
 			c.SSHBastionPrivateKeyFile = c.SSHPrivateKeyFile
-		}
 
-		if c.SSHBastionCertificateFile == "" && c.SSHCertificateFile != "" {
-			c.SSHBastionCertificateFile = c.SSHCertificateFile
+			// Only try to set the bastion certificate to the ssh certificate if the bastion private key isn't set
```

Comment:
I can see a mismatch here if a user provides a BastionCertFile but no BastionKey. In this case the SSHPrivateKeyFile will be used as SSHBastionPrivateKeyFile but the cert will be unmodified. There looks to be a validation check to ensure a bastion key file is set but if that changes it could be an issue. 

```go
		if (c.SSHBastionPrivateKeyFile == "" && c.SSHBastionCertificateFile == "") && c.SSHPrivateKeyFile != "" {
			c.SSHBastionPrivateKeyFile = c.SSHPrivateKeyFile
			c.SSHBastionCertificateFile = c.SSHCertificateFile
		}
```

Reply:
Hm, in the case where someone provides a cert and no key, this should always be an error rather than trying to fallback.
I'll add this case to the test suite and make sure it errors. This should also be the case for the host SSH connection, and there doesn't seem to be any check for this over here, if the host SSH key isn't set, the host cert is ignored.